CMakeLists.txt

```diff
@@ -75,18 +75,32 @@ IF(NOT KOKKOS_HAS_TRILINOS)
   cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
   set(CMAKE_DISABLE_SOURCE_CHANGES ON)
   set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
-  IF (Spack_WORKAROUND)
-    #if we are explicitly using Spack for development,
-    #nuke the Spack compiler
-    SET(SPACK_CXX $ENV{SPACK_CXX})
-    IF(SPACK_CXX)
-      SET(CMAKE_CXX_COMPILER ${SPACK_CXX} CACHE STRING "the C++ compiler" FORCE)
-      SET(ENV{CXX} ${SPACK_CXX})
+
+  # What language are we compiling Kokkos as
+  # Downstream dependencies need to match this!
+  SET(KOKKOS_CMAKE_COMPILE_LANGUAGE CXX)
+  IF (Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE)
+    SET(KOKKOS_INTERNAL_CMAKE_EXTRA_COMPILE_LANGUAGE CXX)
+    IF (Kokkos_ENABLE_CUDA)
+      SET(KOKKOS_CMAKE_COMPILE_LANGUAGE CUDA)
+    ENDIF()
+    IF (Kokkos_ENABLE_HIP)
+      SET(KOKKOS_CMAKE_COMPILE_LANGUAGE HIP)
+    ENDIF()
+  ELSE()
+    IF (Spack_WORKAROUND)
+      #if we are explicitly using Spack for development,
+      #nuke the Spack compiler
+      SET(SPACK_CXX $ENV{SPACK_CXX})
+      IF(SPACK_CXX)
+        SET(CMAKE_CXX_COMPILER ${SPACK_CXX} CACHE STRING "the C++ compiler" FORCE)
+        SET(ENV{CXX} ${SPACK_CXX})
+      ENDIF()
```

Comment:
because our spack pacakge does enable nvcc_wrapper and what not or?
So I think we would need to fix that to really allow this.

Reply:
OK but if we can't use `Spack` with the new option, I think it's more appropriate to say something like
```
IF (Spack_WORKAROUND)
  IF (Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE)
    MESSAGE(FATAL_ERROR "Can't use Kokkos_ENABLE_COMPILER_AS_CMAKE_LANGUAGE in a spack installation!")
  ENDIF()
[...]
ENDIF()
```
Otherwise, we would silently ignore `Spack_WORKAROUND` in case `Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE` is enabled.